test/test.js

```diff
@@ -20,6 +22,24 @@ describe('JSON', function(){
       });
     });
   });
+
+  describe('gocover', function() {
+    it('should return the correct filenames', function(done) {
+      process.env.GOPATH = rootDirectory; // files will be at $GOPATH/src
+      formatter.format(gocoverFixture, function(err, data) {
+        var names = data.source_files.map(function(elem) {
+          return elem.name;
+        });
+        var expected = [
```

Comment:
:+1: (If I make the time/motivation!) I just looked at what subset of the `lcov` output format the formatter used, and tried to generate the same structure.

If another input format was added, I'd consider formalising the data structure format that is passed to CodeClimate (or doubling down on `lcov` output) and splitting the formatter into separate parsing and transformation-to-CodeClimate-API-JSON modules.